web/packages/teleport/src/services/clusters/clusters.ts

```diff
@@ -21,10 +21,19 @@ import cfg from 'teleport/config';
 
 import { makeClusterList } from './makeCluster';
 
-const service = {
-  fetchClusters() {
-    return api.get(cfg.api.clustersPath).then(makeClusterList);
-  },
-};
+import { Cluster } from '.';
+
+export default class ClustersService {
+  clusters: Cluster[] = [];
 
-export default service;
+  fetchClusters() {
+    if (this.clusters.length < 1) {
+      return api.get(cfg.api.clustersPath).then(res => {
+        // cache the result of clusters response
+        this.clusters = makeClusterList(res);
+        return this.clusters;
+      });
+    }
+    return Promise.resolve(this.clusters);
+  }
```

Comment:
i think we should give it the option to fetch a fresh batch of clusters. i think we do something similar with `fetchUserContext` where if we want to read from cache, pass `fromCache = true`